### ISSUE TITLE
Provider produced invalid plan

### ISSUE THREAD

**Issue by ChrisPortman** [hashicorp/terraform-provider-local#189]
### Terraform CLI and Provider Versions

Terraform v1.4.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.76.1
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/tls v4.0.4

### Terraform Configuration

```terraform
locals {
  connection_details = [ for i, v in var.hosts: {
    ip_address = module.host[i].ip_address
  }]

  connection_details_json = jsonencode(local.connection_details)
}

resource "local_file" "connection_details" {
  content = local.connection_details_json
  filename = "${var.work_dir}/connection_details"
  file_permission = "0644"
}
```

### Expected Behavior

It should create a file with the json encoded representation of the data that includes a value that is an output of a module (not sure if that's relevant).

### Actual Behavior

Error: Provider produced invalid plan

Provider "registry.terraform.io/hashicorp/local" planned an invalid value for local_file.connection_details.content: planned value cty.UnknownVal(cty.String) does not match config value
cty.UnknownVal(cty.String).Refine().NotNull().StringPrefixFull("[").NewValue().

This is a bug in the provider, which should be reported in the provider's own issue tracker.

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

Medium

### Additional Information

Rolling the provider back to version 2.2.3 gets me around this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by austinvalle** [comment]
Hi there @ChrisPortman 👋🏻, thanks for the report and sorry that you're running into this issue. 

This looks similar to [a reported bug](https://github.com/hashicorp/terraform/issues/32859) on Terraform itself, version `1.4.x`. That issue is currently being investigated by the core team, but in the meantime, you should be able to downgrade your terraform version to `1.3.9` as well to workaround that issue: https://releases.hashicorp.com/terraform/1.3.9/ 

I will keep this thread updated on the status of that bug fix 👍🏻

**Comment by austinvalle** [comment]
That issue has been fixed with the release of Terraform `1.4.2`:
- https://github.com/hashicorp/terraform/blob/v1.4.2/CHANGELOG.md#142-march-16-2023 
- https://releases.hashicorp.com/terraform/1.4.2/
